### DISCUSSION TITLE
Custom Fields - Send files in the contact form.

### DISCUSSION THREAD

**Discussion by MrsHeadcrash** [joomla/joomla-cms#17936] [2. Cold Features]
### Steps to reproduce the issue
Unfortunately, I have found nothing suitable and should make this request already exist - please delete this.

If you want to install a custom field in a contact form or similar, there is unfortunately not the possibility to create a field by sending files.

Would it be possible to include this?
-State the file size
-which file format can be sent.
-Selection of one or more files.

thx

**Comment by coolcat-creations** [comment]
A general file upload field without access to the media manager in core would be indeed great.
I can see a lot of use cases such as the user avatar or like written in the contact form.

+1 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17936">issues.joomla.org/tracker/joomla-cms/17936</a>.</sub>

**Comment by mbabker** [comment]
A general purpose file upload field isn't practical (even with "normal" forms), there is a lot of extra logic that has to be wired up to process the form submission and specifically the uploads that makes it pretty difficult to create a general purpose solution.  File uploads are generally something you need to have something specific to your workflow to use.

**Comment by MrsHeadcrash** [comment]
in the first line it is me that I get pictures sent. of course it could be other file formats. Why should not that be practical? It always depends on for which purpose this is used. This should then perhaps be selected via a selection field. Which internal networks there must be .. no idea. I ask the question to the professionals. if this is not possible, then that is ok. if the effort is not worth it, then this is also ok.

- thank you - to google for the translation

**Comment by mbabker** [comment]
File inputs are probably the most trickiest of the HTML input types because they require you to write a server side processor for that data specifically.  The logic processing the data has to read the file upload data, perform validation rules on it (based on your form's configuration as well as basic security checks), and handle putting it into the filesystem in the desired location.  A lot of that upload logic is going to be very specific to a use case, so it's difficult to write a single plugin that just handles all types of file uploads with an unlimited number of constraints/configurations.

The contact form is an even more unique case as if you're looking for the email that gets sent out to have the upload(s) as attachments then the custom field logic has to be able to hook into the email sending logic; there isn't an integration point for that right now.

Not saying that this absolutely cannot be done, but to make a generic upload custom field type takes a lot more planning and consideration than most other field types.

**Comment by MrsHeadcrash** [comment]
thank you for your information, then I must look for a different solution

**Comment by pepperstreet** [comment]
+1
Hello, what about Allon Moritz's DPAttachments? Maybe his solution is a possible candidate for a similar core feature? Would be nice to hear some thoughts from master @laoneo ;)

Besides, what about the up-coming new media-manager? Doesn't it have an accompanying media field and more flexible features? For instance: Restrict uploads and formats per field, restrict directories, restrict UI display ... Hopefully a better ACL and fine-grained permissions.

[a comment by an automated account is not shown]

**Comment by ghost** [comment]
closed as following the [Comment](https://github.com/joomla/joomla-cms/issues/17936#issuecomment-328598092 "") of @mbabker <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17936">issues.joomla.org/tracker/joomla-cms/17936</a>.</sub>

**Comment by laoneo** [comment]
ACL in media manager needs to be discussed if we really want to got that way as it requires a full blown document manager solution which we have already extensions which are doing it. DPAttachments would also not fit for this use case and I have no plans to migrate that into core as extensions should do jobs like that.

**Comment by coolcat-creations** [comment]
Why is it already closed @franz-wohlkoenig ? @mbabker said also "Not saying that this absolutely cannot be done"
Wouldn't it be good to provide a field from core to prevent others doing it the wrong way? :)

[a comment by an automated account is not shown]

**Comment by ghost** [comment]
reopened as stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17936">issues.joomla.org/tracker/joomla-cms/17936</a>.</sub>

**Comment by Bakual** [comment]
The problem with such a field is that it requires security checks and heavy server side processing. That processing would be very situational as to what should be allowed and where to store the uploaded files.
This likely needs a component dealing with that.

Imho, we shouldn't add a generic custom field for file upload. This is one of the cases where 3rd party extensions can jump in and provide specialised solutions for the various use cases.

**Comment by dgrammatiko** [comment]
I think this issue should be against the new media manager (NMM) repo.
**If** media manager is always the component that deals with all the user related files (I guess this is the aim for the NMM) then this could be possible tackled is some way there (I mean beefing up the media field with conjunction to the fields params and the storage adapters).
I can think a way that this could work, but obviously it needs some testing/experimentation etc..

**Comment by denvarel** [comment]
Any news about a Upload File - Custom Field?

**Comment by Gitjk** [comment]
>Any news about a Upload File - Custom Field?

I've recently attempted to produce a website showcase with Joomla custom fields, where users of a certain component can register and add a description of their live website from the frontend. One of the problems I ran into is that I can't limit an author to see only his/her own homepage screenshot in the media manager.

However, there is a new extension (I didn't test it yet) which might fill the shortcomings of using Joomla custom fields for such custom field applications. See: [Field JT Fileupload](https://extensions.joomla.org/extensions/extension/authoring-a-content/custom-fields/field-jt-fileupload/ "Field JT Fileupload") Currently it is limited to uploading pdf files, but I think it's pretty easy to add more file types.


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17936">issues.joomla.org/tracker/joomla-cms/17936</a>.</sub>
